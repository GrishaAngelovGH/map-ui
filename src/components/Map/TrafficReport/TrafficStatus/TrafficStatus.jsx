const statuses = {
  low: {
    label: 'Low Traffic',
    classes: 'alert alert-warning'
  },
  normal: {
    label: 'Normal Traffic',
    classes: 'alert alert-success'
  },
  high: {
    label: 'High Traffic',
    classes: 'alert alert-danger'
  }
}

const TrafficStatus = ({ status }) => (
  <div className='row mt-3 justify-content-center'>
    <div className='col-md-10 border rounded shadow'>

      <div className='row small'>
        <div className='col-md-4 p-3'>
          <div className={`${statuses[status].classes} m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold`}>
            <p className='m-0'>9 min</p>
            <p className='m-0'>(1.9 km)</p>
          </div>
        </div>

        <div className='col-md-8 p-1'>
          <p className='m-0 text-success text-center fw-bold'>Via A4 and A308</p>
          <p className={`${statuses[status].classes} text-center fw-bold m-0 p-1`}>{statuses[status].label}</p>
          <p className='m-0 text-secondary d-flex align-items-center'>
            <i className='bi bi-sign-turn-slight-right-fill text-primary me-2'></i>
            Next Direction
          </p>
          <p className='m-0'>Head west on Knightsbridge</p>
        </div>
      </div>

    </div>
  </div>
)

export default TrafficStatus