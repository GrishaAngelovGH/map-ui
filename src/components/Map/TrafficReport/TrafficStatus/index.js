import TrafficStatus from './TrafficStatus'

export default TrafficStatus