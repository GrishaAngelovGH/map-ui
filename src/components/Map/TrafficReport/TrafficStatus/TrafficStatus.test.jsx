import { render } from '@testing-library/react'

import TrafficStatus from './TrafficStatus'

test('should render low TrafficStatus component', () => {
  const view = render(<TrafficStatus status='low' />)

  expect(view).toMatchSnapshot()
})

test('should render normal TrafficStatus component', () => {
  const view = render(<TrafficStatus status='normal' />)

  expect(view).toMatchSnapshot()
})

test('should render high TrafficStatus component', () => {
  const view = render(<TrafficStatus status='high' />)

  expect(view).toMatchSnapshot()
})