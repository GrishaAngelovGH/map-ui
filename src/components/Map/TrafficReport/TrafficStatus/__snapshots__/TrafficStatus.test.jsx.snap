// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render high TrafficStatus component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mt-3 justify-content-center"
      >
        <div
          class="col-md-10 border rounded shadow"
        >
          <div
            class="row small"
          >
            <div
              class="col-md-4 p-3"
            >
              <div
                class="alert alert-danger m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
              >
                <p
                  class="m-0"
                >
                  9 min
                </p>
                <p
                  class="m-0"
                >
                  (1.9 km)
                </p>
              </div>
            </div>
            <div
              class="col-md-8 p-1"
            >
              <p
                class="m-0 text-success text-center fw-bold"
              >
                Via A4 and A308
              </p>
              <p
                class="alert alert-danger text-center fw-bold m-0 p-1"
              >
                High Traffic
              </p>
              <p
                class="m-0 text-secondary d-flex align-items-center"
              >
                <i
                  class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                />
                Next Direction
              </p>
              <p
                class="m-0"
              >
                Head west on Knightsbridge
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mt-3 justify-content-center"
    >
      <div
        class="col-md-10 border rounded shadow"
      >
        <div
          class="row small"
        >
          <div
            class="col-md-4 p-3"
          >
            <div
              class="alert alert-danger m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
            >
              <p
                class="m-0"
              >
                9 min
              </p>
              <p
                class="m-0"
              >
                (1.9 km)
              </p>
            </div>
          </div>
          <div
            class="col-md-8 p-1"
          >
            <p
              class="m-0 text-success text-center fw-bold"
            >
              Via A4 and A308
            </p>
            <p
              class="alert alert-danger text-center fw-bold m-0 p-1"
            >
              High Traffic
            </p>
            <p
              class="m-0 text-secondary d-flex align-items-center"
            >
              <i
                class="bi bi-sign-turn-slight-right-fill text-primary me-2"
              />
              Next Direction
            </p>
            <p
              class="m-0"
            >
              Head west on Knightsbridge
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render low TrafficStatus component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mt-3 justify-content-center"
      >
        <div
          class="col-md-10 border rounded shadow"
        >
          <div
            class="row small"
          >
            <div
              class="col-md-4 p-3"
            >
              <div
                class="alert alert-warning m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
              >
                <p
                  class="m-0"
                >
                  9 min
                </p>
                <p
                  class="m-0"
                >
                  (1.9 km)
                </p>
              </div>
            </div>
            <div
              class="col-md-8 p-1"
            >
              <p
                class="m-0 text-success text-center fw-bold"
              >
                Via A4 and A308
              </p>
              <p
                class="alert alert-warning text-center fw-bold m-0 p-1"
              >
                Low Traffic
              </p>
              <p
                class="m-0 text-secondary d-flex align-items-center"
              >
                <i
                  class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                />
                Next Direction
              </p>
              <p
                class="m-0"
              >
                Head west on Knightsbridge
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mt-3 justify-content-center"
    >
      <div
        class="col-md-10 border rounded shadow"
      >
        <div
          class="row small"
        >
          <div
            class="col-md-4 p-3"
          >
            <div
              class="alert alert-warning m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
            >
              <p
                class="m-0"
              >
                9 min
              </p>
              <p
                class="m-0"
              >
                (1.9 km)
              </p>
            </div>
          </div>
          <div
            class="col-md-8 p-1"
          >
            <p
              class="m-0 text-success text-center fw-bold"
            >
              Via A4 and A308
            </p>
            <p
              class="alert alert-warning text-center fw-bold m-0 p-1"
            >
              Low Traffic
            </p>
            <p
              class="m-0 text-secondary d-flex align-items-center"
            >
              <i
                class="bi bi-sign-turn-slight-right-fill text-primary me-2"
              />
              Next Direction
            </p>
            <p
              class="m-0"
            >
              Head west on Knightsbridge
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`should render normal TrafficStatus component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row mt-3 justify-content-center"
      >
        <div
          class="col-md-10 border rounded shadow"
        >
          <div
            class="row small"
          >
            <div
              class="col-md-4 p-3"
            >
              <div
                class="alert alert-success m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
              >
                <p
                  class="m-0"
                >
                  9 min
                </p>
                <p
                  class="m-0"
                >
                  (1.9 km)
                </p>
              </div>
            </div>
            <div
              class="col-md-8 p-1"
            >
              <p
                class="m-0 text-success text-center fw-bold"
              >
                Via A4 and A308
              </p>
              <p
                class="alert alert-success text-center fw-bold m-0 p-1"
              >
                Normal Traffic
              </p>
              <p
                class="m-0 text-secondary d-flex align-items-center"
              >
                <i
                  class="bi bi-sign-turn-slight-right-fill text-primary me-2"
                />
                Next Direction
              </p>
              <p
                class="m-0"
              >
                Head west on Knightsbridge
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row mt-3 justify-content-center"
    >
      <div
        class="col-md-10 border rounded shadow"
      >
        <div
          class="row small"
        >
          <div
            class="col-md-4 p-3"
          >
            <div
              class="alert alert-success m-0 h-100 d-flex flex-column justify-content-center align-items-center fw-bold"
            >
              <p
                class="m-0"
              >
                9 min
              </p>
              <p
                class="m-0"
              >
                (1.9 km)
              </p>
            </div>
          </div>
          <div
            class="col-md-8 p-1"
          >
            <p
              class="m-0 text-success text-center fw-bold"
            >
              Via A4 and A308
            </p>
            <p
              class="alert alert-success text-center fw-bold m-0 p-1"
            >
              Normal Traffic
            </p>
            <p
              class="m-0 text-secondary d-flex align-items-center"
            >
              <i
                class="bi bi-sign-turn-slight-right-fill text-primary me-2"
              />
              Next Direction
            </p>
            <p
              class="m-0"
            >
              Head west on Knightsbridge
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
